### PR TITLE
move aruco from contrib to objdetect in main repository

### PR DESCRIPTION
**Pull request by AleksandrPanov**
1. Move aruco_detector, aruco_board, aruco_dictionary, aruco_utils to objdetect
2. Reduce inludes
3. Fix java tests (add objdetect import)
4. Refactoring
5. Reduce aruco pose estimate to fix refineDetectedMarkers

merge with https://github.com/opencv/opencv/pull/22368

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

**Review by asmorkalov — APPROVED** [review]
:+1:

**Comment by stefan523** [comment]
Is there any chance to merge this with pull request "https://github.com/opencv/opencv_contrib/pull/3305" / issue "https://github.com/opencv/opencv_contrib/issues/3291"?

File: modules/aruco/include/opencv2/aruco/charuco.hpp [hunk 1]
```diff
@@ -90,7 +90,8 @@ CV_EXPORTS_W void detectCharucoDiamond(InputArray image, InputArrayOfArrays mark
                                        OutputArrayOfArrays diamondCorners, OutputArray diamondIds,
                                        InputArray cameraMatrix = noArray(),
                                        InputArray distCoeffs = noArray(),
-                                       Ptr<Dictionary> dictionary = getPredefinedDictionary(PREDEFINED_DICTIONARY_NAME::DICT_4X4_50));
+                                       Ptr<Dictionary> dictionary = makePtr<Dictionary>
```
Comment (asmorkalov) [inline modules/aruco/include/opencv2/aruco/charuco.hpp:93]: Why we need Ptr here?

File: modules/aruco/samples/detect_diamonds.cpp [hunk 2]
```diff
@@ -166,7 +166,7 @@ int main(int argc, char *argv[]) {
         vector< Vec3d > rvecs, tvecs;
 
         // detect markers
-        aruco::detectMarkers(image, dictionary, markerCorners, markerIds, detectorParams,
+        aruco::detectMarkers(image, makePtr<aruco::Dictionary>(dictionary), markerCorners, markerIds, detectorParams,
```
Comment (asmorkalov) [inline modules/aruco/samples/detect_diamonds.cpp:169]: The same question here.

**Review by asmorkalov — APPROVED** [review]

**Pull request merged**